packages/api/src/APIProvider.tsx

```diff
@@ -118,15 +119,38 @@ const initializeDerivAPI = (): DerivAPIBasic => {
 
 const queryClient = getSharedQueryClientContext();
 
+/**
+ * Determines the WS environment based on the login ID and custom server URL.
+ * @param {string | null | undefined} loginid - The login ID (can be a string, null, or undefined).
+ * @returns {string} Returns the WS environment: 'custom', 'real', or 'demo'.
+ */
+const getEnvironment = (loginid: string | null | undefined) => {
+    const customServerURL = window.localStorage.getItem('config.server_url');
+    if (customServerURL) return 'custom';
+
+    if (loginid && !/^(VRT|VRW)/.test(loginid)) return 'real';
+    return 'demo';
+};
+
 type TAPIProviderProps = {
     /** If set to true, the APIProvider will instantiate it's own socket connection. */
     standalone?: boolean;
 };
 
 const APIProvider = ({ children, standalone = false }: PropsWithChildren<TAPIProviderProps>) => {
     const WS = useWS();
+    const activeLoginid = window.localStorage.getItem('active_loginid');
+    const [environment, setEnvironment] = useState(getEnvironment(activeLoginid));
     const standaloneDerivAPI = useRef(standalone ? initializeDerivAPI() : null);
 
+    const switchEnvironment = (loginid: string | null | undefined) => {
```

Comment:
whats "environement"? is it always derived from login? should it be in sync with login?

Reply:
yes, if not defaults to demo environment. it is to switch between WS endpoints (blue/green.derivws.com) - real/demo. qa servers (custom)

Reply:
yes~